backend/typescript/rest/givecloud.ts

```diff
@@ -0,0 +1,43 @@
+import { Router } from "express";
+import AuthService from "../services/implementations/authService";
+import UserService from "../services/implementations/userService";
+import { sendErrorResponse } from "../utilities/errorResponse";
+import { Role } from "../types";
+
+const givecloudRouter: Router = Router();
+
+givecloudRouter.post("/", async (req, res) => {
+  const { membership } = req.body;
+  const { supporter } = req.body;
+  const date = new Date();
+
+  const userService = new UserService();
+  const authService = new AuthService(userService);
+
+  try {
+    const { email } = supporter;
+    let roleType: Role = "Subscriber";
+    if (membership.name === "Professional Creator Membership") {
+      roleType = "Author";
+    }
+    if ((await userService.getUserByEmail(email)) == null) {
+      date.setDate(date.getDate() + membership.days_to_expire);
+      const user = await userService.createUser({
+        firstName: supporter.first_name,
+        lastName: supporter.last_name,
+        email: supporter.email,
+        roleType,
+        subscriptionExpiresOn: date,
+        password: "",
+      });
+      await authService.sendEmailVerificationLink(email);
```

Comment:
We don't actually use `sendEmailVerificationLink`. This is a holdover from starter code. We've implemented a custom `sendPasswordSetupLink` instead for users to setup their account. 

See the `/register` route for more details. We basically want to replicate that signup flow here, but with the non-Admin roletypes. You may find it useful to move that code into a separate function (Perhaps in the user service? idk) and just call it here.